Sources/Instrumentation/Instrument.swift

```diff
@@ -14,10 +14,21 @@
 
 import InstrumentationBaggage
 
+/// Typealias used to simplify Support of old Swift versions which do not have `Sendable` defined.
+#if swift(>=5.6.0)
+@preconcurrency public protocol _SwiftInstrumentationSendable: Sendable {}
+#else
+public protocol _SwiftInstrumentationSendable {}
+#endif
+
 /// Conforming types are used to extract values from a specific `Carrier`.
-public protocol Extractor {
+public protocol Extractor: _SwiftInstrumentationSendable {
     /// The carrier to extract values from.
+    #if swift(>=5.6.0)
+    associatedtype Carrier: Sendable
```

Comment:
Can't we use the `_SwiftInstrumentationSendable` protocol as the associated type constraint?

Reply:
That would force people to conform to `_SwiftInstrumentationSendable`, but we just need to force them to conform to `Sendable` 💡

Reply:
Aah, makes sense 👍